### PR TITLE
[Docs] Add Migration guide + some API updates

### PR DESCRIPTION
**Pull request by P-R-O-C-H-Y**
## Description of Change
Created Migration guide page in documentation for migration from 2.x to 3.0.
Updated SigmaDelta API docs.
Also added to some peripherals docs informations about C6/H2 + some minor fixes.

DRAFT:
- missing RMT
- missing UART

### CODE REVIEW

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+Removed APIs
+************
+
+* ``hallRead``
+
+I2S
+---
+
+I2S driver has beed completely redesigned and refactored to use new ESP-IDF driver. 
+For more informations about the new API check `I2S API <https://docs.espressif.com/projects/arduino-esp32/en/latest/api/i2s.html>`_.
```
[the remark this answers is not in the document]
Comment (P-R-O-C-H-Y) [reply docs/source/migration_guide/2.x_to_3.0.rst:36]: Good catch, will rework that. Thanks

**Comment by PilnyTomas** [comment]
Could you please mention changes in BLE? ... assuming it will be merged and available in 3.0
https://github.com/espressif/arduino-esp32/pull/7867

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+************
+
+* ``ledcSetup``
+* ``ledcAttachPin``
+
+New APIs
+********
+
+* ``ledcAttach`` used to setup LEDC pin (merged ``ledcSetup`` and ``ledcAttachPin`` functions).
+* ``timerGetFrequency`` used to get actual frequency of the timer.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:62]:
```suggestion
* ``timerGetFrequency`` used to get the actual frequency of the timer.
```

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+* ``timerAttachInterruptArg`` used to attach interrupt to timer using argument.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:63]:
```suggestion
* ``timerAttachInterruptArg`` used to attach the interrupt to a timer using arguments.
```

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+---
+
+I2S driver has beed completely redesigned and refactored to use new ESP-IDF driver. 
+For more informations about the new API check `I2S API <https://docs.espressif.com/projects/arduino-esp32/en/latest/api/i2s.html>`_.
+
+LEDC
+----
+
+LEDC API has been changed in order to support the Peripheral Manager and easier use as LEDC channels are now automaticaly assigned to pins.
+For more informations about the new API check `LEDC API <https://docs.espressif.com/projects/arduino-esp32/en/latest/api/ledc.html>`_.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:50]:
```suggestion
For more information about the new API, check `LEDC API <https://docs.espressif.com/projects/arduino-esp32/en/latest/api/ledc.html>`_.
```

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+Removed APIs
+************
+
+* ``sigmaDeltaSetup``
+* ``sigmaDeltaRead``
+
+New APIs
+********
+
+* ``sigmaDeltaAttach`` used to setup SigmaDelta pin (channel is aquired automatically).
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:107]:
```suggestion
* ``sigmaDeltaAttach`` used to set up the SigmaDelta pin (channel is acquired automatically).
```

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+* ``timerGetFrequency`` used to get actual frequency of the timer.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:108]:
```suggestion
* ``timerGetFrequency`` used to get the actual frequency of the timer.
```

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+* ``timerAttachInterruptArg`` used to attach interrupt to timer using argument.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:109]:
```suggestion
* ``timerAttachInterruptArg`` used to attach the interrupt to a timer using arguments.
```

File: docs/source/api/sigmadelta.rst [hunk 8]
```diff
@@ -14,67 +14,61 @@ ESP32 SoC Number of SigmaDelta channels
...
+This function will return ``true`` if setting duty is successful.
+If ``false`` is returned, error occurs and duty was not set.
 
-sigmaDeltaDetachPin
-*******************
+sigmaDeltaDetach
+****************
 
-This function is used to detach pin from SigmaDelta.
+This function is used to detach pin from SigmaDelta and deinit the channel that was attached to the pin.
```
Comment (lucasssvaz) [inline docs/source/api/sigmadelta.rst:61]:
```suggestion
This function is used to detach a pin from SigmaDelta and deinit the channel that was attached to the pin.
```

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 9]
```diff
@@ -0,0 +1,176 @@
+#########################
+Migration from 2.x to 3.0
+#########################
+
+Introduction
+------------
+
+This is a guide to show breaking changes in API and to help migrate projects from 2.X version to 3.0 version of Arduino ESP32 core. 
+For more informations about all changes and new features check `RELESE NOTES <https://github.com/espressif/arduino-esp32/releases>`_.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:9]:
```suggestion
For more information about the changes and new features, check `RELEASE NOTES <https://github.com/espressif/arduino-esp32/releases>`_.
```

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+Removed APIs
+************
+
+* ``hallRead``
+
+I2S
+---
+
+I2S driver has beed completely redesigned and refactored to use new ESP-IDF driver. 
+For more informations about the new API check `I2S API <https://docs.espressif.com/projects/arduino-esp32/en/latest/api/i2s.html>`_.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:44]:
```suggestion
For more information about the new API, check `I2S API <https://docs.espressif.com/projects/arduino-esp32/en/latest/api/i2s.html>`_.
```

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+***************
+
+* ``ledcDetachPin`` renamed to ``ledcDetach``.
+* In all functions input parameter ``channel`` have been changed to ``pin``.
+
+RMT
+---
+
+TODO:
+For more informations about the new API check `RMT API <https://docs.espressif.com/projects/arduino-esp32/en/latest/api/rmt.html>`_.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:75]:
```suggestion
For more information about the new API, check `RMT API <https://docs.espressif.com/projects/arduino-esp32/en/latest/api/rmt.html>`_.
```

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+Changes in APIs
+***************
+
+TODO:
+
+SigmaDelta
+----------
+
+SigmaDelta has been refactored to use new ESP-IDF driver.
+For more informations about the new API check `SigmaDelta API <https://docs.espressif.com/projects/arduino-esp32/en/latest/api/sigmadelta.html>`_.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:96]:
```suggestion
For more information about the new API, check `SigmaDelta API <https://docs.espressif.com/projects/arduino-esp32/en/latest/api/sigmadelta.html>`_.
```

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+* ``timerAlarm`` used to setup Alarm for timer and enables it automaticaly (merged ``timerAlarmWrite`` and ``timerAlarmEnable`` functions).
+* ``timerGetFrequency`` used to get actual frequency of the timer.
+* ``timerAttachInterruptArg`` used to attach interrupt to timer using argument.
+
+Changes in APIs
+***************
+
+* ``timerBegin`` now have only 1 parameter wich is frequency. There is an automatic calculation of divider through different clock sources¨
+  to achieve selected frequency.
+* ``timerAttachInterrupt`` now have only 2 parameters -> parameter ``edge`` has been removed.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:155]:
```suggestion
* ``timerBegin`` has now only 1 parameter (frequency). There is an automatic calculation of the divider using different clock sources
  to achieve the selected frequency.
* ``timerAttachInterrupt`` has now only 2 parameters. The ``edge`` parameter has been removed.
```

File: docs/source/migration_guide/2.x_to_3.0.rst [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+
+UART
+----
+
+TODO:
+For more informations about the new API check `UART API <https://docs.espressif.com/projects/arduino-esp32/en/latest/api/rmt.html>`_.
```
Comment (lucasssvaz) [inline docs/source/migration_guide/2.x_to_3.0.rst:161]:
```suggestion
For more information about the new API, check `UART API <https://docs.espressif.com/projects/arduino-esp32/en/latest/api/rmt.html>`_.
```

**Review by lucasssvaz — COMMENTED** [review]
My suggestions are mostly for making the text cleaner. I'll keep reviewing as more things are added

**Comment by VojtechBartoska** [comment]
@SuGlider please add later sections about UART and RMT, thanks!

**Comment by P-R-O-C-H-Y** [comment]
Closing - rebased PR created #8716

**Pull request closed without merging**